### PR TITLE
{phys}[foss/2022a] HOOMD-blue v4.0.1 w/ Python 3.10.4

### PR DESCRIPTION
**Pull request by akesandgren**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
b-cn1606.hpc2n.umu.se - Linux Ubuntu 22.04, x86_64, AMD EPYC 7763 64-Core Processor, Python 3.10.6
See https://gist.github.com/akesandgren/691c05bcb619a9012dbe49106abeb593 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
b-an02.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 3.8.10
See https://gist.github.com/akesandgren/b8902430e90d2c09b4b573060dc99c78 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
bear-pg0105u03a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/b0e6e824d6d241fc7397d1ed8bc27802 for a full test report.

**Comment by akesandgren** [comment]
What is going o nhere??

I see no problems whatsoever in my builds...

Is it picking up something from the OS that it shouldn't?

**Comment by akesandgren** [comment]
@branfosj what was actually failing i nyour test of this one?

**Comment by branfosj** [comment]
> What is going o nhere??
> 
> I see no problems whatsoever in my builds...
> 
> Is it picking up something from the OS that it shouldn't?

I've found the issue. It is picking up the Python from my EasyBuild virtualenv instead of the Python from the dependency.

```
-- Found PythonInterp: /rds/projects/2017/branfosj-rse/easybuild/venv-dev-EL8/bin/python (found suitable version "3.6.8", minimum required is "3")
-- Found PythonLibs: /usr/lib64/libpython3.6m.so
```

Neither of `-DPython_EXECUTABLE=$EBROOTPYTHON/bin/python` or `-DPython_ROOT_DIR=$EBROOTPYTHON` has fixed it.

**Comment by akesandgren** [comment]
Ok, let me check...

**Comment by akesandgren** [comment]
Try setting -DPYTHON_EXECUTABLE=.... this comes from CMake/3.24's deprecated FindPythonInterp

File: easybuild/easyconfigs/h/HOOMD-blue/HOOMD-blue-4.0.1-foss-2022a.eb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+]
+
+_copts = [
+    '-DENABLE_GPU=OFF',
+    '-DENABLE_MPI=ON',
+    '-DBUILD_MD=ON',
+    '-DBUILD_METAL=ON',
+    '-DENABLE_TBB=ON',
+    '-DBUILD_TESTING=ON',
+]
```
Comment (branfosj) [inline easybuild/easyconfigs/h/HOOMD-blue/HOOMD-blue-4.0.1-foss-2022a.eb:39]:
```suggestion
    '-DPYTHON_EXECUTABLE=$EBROOTPYTHON/bin/python',
]
```

**Comment by branfosj** [comment]
@boegelbot please test @ generoso

**Review by branfosj — APPROVED** [review]

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0105u03a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/26fdaa24ec8d29ded5985f97d2d17d35 for a full test report.

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen2

**Comment by branfosj** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

File: easybuild/easyconfigs/h/HOOMD-blue/HOOMD-blue-4.0.1-foss-2022a.eb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    '-DENABLE_TBB=ON',
+    '-DBUILD_TESTING=ON',
+    '-DPYTHON_EXECUTABLE=$EBROOTPYTHON/bin/python',
+]
+configopts = ' '.join(_copts)
+
+postinstallcmds = [
+    'ln -s hoomd/include %(installdir)s/include',
+]
+
```
Comment (branfosj) [inline easybuild/easyconfigs/h/HOOMD-blue/HOOMD-blue-4.0.1-foss-2022a.eb:46]:
```suggestion

pretestopts = "OMPI_MCA_rmaps_base_oversubscribe=1 "
```

**Review by branfosj — APPROVED** [review]

**Comment by branfosj** [comment]
@boegelbot please test @ generoso

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen2

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0105u03a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/9b903045a5d8460792b5f254170e19a7 for a full test report.

**Comment by branfosj** [comment]
Going in, thanks @akesandgren!

**Pull request merged**